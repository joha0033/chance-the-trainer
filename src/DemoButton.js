import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';

const styles = theme => ({
    root: {
        width: '100%',
        maxWidth: 450,
        backgroundColor: theme.palette.background.paper,
      },
    button: {
        color: 'white',
    },
    icon:{
        paddingRight: 10
    }
  
});

const handleClick = (setIsDemoDisplayed) => {
    return setIsDemoDisplayed()
}


const DemoButton = (props) => {
  const { classes, setIsDemoDisplayed } = props;
  return (
    <div className='ProgressBar'>
        <div >
            <Button 
                aria-label="Demo" 
                className={classes.button}
                onClick={() => handleClick(setIsDemoDisplayed)}
                // style={{padding: progressUsed * 1}}
            >
            <i className={"material-icons " + classes.icon}>
            burst_mode
            </i>
                demo
            </Button>
        </div>
      </div>
  );
}

DemoButton.propTypes = {
  classes: PropTypes.object.isRequired,
};


export default withStyles(styles)(DemoButton)