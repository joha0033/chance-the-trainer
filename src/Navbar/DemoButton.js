import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';

const styles = theme => ({
    root: {
        width: '100%',
        maxWidth: 450,
        backgroundColor: theme.palette.background.paper,
      },
    button: {
        color: 'white',
        width: '160px'
    },
    icon:{
        paddingRight: 10
    }
  
});




const DemoButton = (props) => {
    const { classes } = props;
    const [isDemoDisplayed, setIsDemoDisplayed] = useState(false)

    const handleClick = () => {
        setIsDemoDisplayed(!isDemoDisplayed)
    }

    return (
        <div className='ProgressBar'>
            <div >
                <label>
                    <Button 
                        aria-label="Demo" 
                        className={classes.button}
                        onClick={() => handleClick()}
                    >
                        <i className={"material-icons " + classes.icon}>
                            burst_mode
                        </i>
                        {isDemoDisplayed ? 'Hide Demo' : 'Show Demo'}
                    </Button>
                    
                </label>
                
            </div>
        </div>
    );
}

DemoButton.propTypes = {
  classes: PropTypes.object.isRequired,
};


export default withStyles(styles)(DemoButton)