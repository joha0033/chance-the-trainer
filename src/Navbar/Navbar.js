import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
// import DemoButton from './DemoButton';
import { Link } from 'react-router-dom'

const styles = theme => ({
    root: {
        flexGrow: 1,
    },
    demoButton: {
      [theme.breakpoints.down('xs')]: {
        display: 'none',
      },
    },
    nameSmall: {
      display: 'none',
      [theme.breakpoints.down('xs')]: {
        display: 'block',
        textAlign: 'center'
      },
    },
    nameLarge: {
      display: 'block',
      [theme.breakpoints.down('xs')]: {
        display: 'none'
      },
    },
    link: { 
      textDecoration: 'none', 
      color: 'white' 
    },
  });

function SimpleAppBar(props) {
  const { classes } = props;

  return (
    <div className={classes.root}>
      <AppBar position="static" color="primary">
        <Toolbar>
            <Grid container spacing={0}>
           
            
                <Grid item xs={12}>
                <Link to='/' className={classes.link}>
                    <Typography variant="h4" color="inherit" className={classes.nameLarge}>
                        Chance The Trainer
                    </Typography>
                    <Typography variant="subtitle2" color="inherit" className={classes.nameSmall}>
                        Chance The Trainer sm
                    </Typography>
                    </Link>
                </Grid>
                <Grid item xs={12}>
                    <Typography variant="h6" color="inherit" className={classes.nameLarge}>
                        by: ajDevs
                    </Typography>
                    <Typography variant="caption" color="inherit" className={classes.nameSmall}>
                        by: ajDevs sm
                    </Typography>
                </Grid>
                
            </Grid>
            {/* <div className={classes.demoButton}>
              <DemoButton />
            </ div> */}
           
        </Toolbar>
      </AppBar>
    </div>
  );
}

SimpleAppBar.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(SimpleAppBar);