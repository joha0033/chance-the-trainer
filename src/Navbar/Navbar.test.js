import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import {render, fireEvent, cleanup, text} from 'react-testing-library';
import Navbar from './Navbar';
import DemoButton from './DemoButton'

/**
 * RENDER + SNAPSHOT TESTS
 * 
 * START
 */

/**
* RENDERS START
*/
//NAVBAR 
it('renders Navbar component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

// DEMO BUTTON
it('renders demo button without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DemoButton />, div);
  ReactDOM.unmountComponentAtNode(div);
});
/**
* RENDERS END
*/


/**
 * SNAPSHOTS START
 */
// NAVBAR
it('renders Navbar component correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// DEMO BUTTON
it('renders demo button correctly', () => {
  const tree = renderer
    .create(<DemoButton />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
/**
 * SNAPSHOTS END
 */

 /**
  * END
  * 
  * RENDER + SNAPSHOT TESTS 
  */



 /**
  * DOM && EVENTS
  * 
  * START
  */
//  afterEach(cleanup);

//  it('Demo button text toggles "Hide Demo" to "Show Demo" on click', () => {
//   const {queryByText, getByText} = render(
//     <DemoButton  />,
//   );

//   expect(queryByText('Show Demo')).toBeTruthy();

//   fireEvent.click(getByText('Show Demo'));

//   expect(queryByText('Hide Demo')).toBeTruthy();
// });
 /**
  * END
  * 
  * DOM && EVENTS
  */