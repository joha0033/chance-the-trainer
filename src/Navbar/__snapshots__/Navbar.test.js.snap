// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar component correctly 1`] = `
<div
  className="SimpleAppBar-root-182"
>
  <header
    className="MuiPaper-root-196 MuiPaper-elevation4-202 MuiAppBar-root-187 MuiAppBar-positionStatic-191 MuiAppBar-colorPrimary-194"
  >
    <div
      className="MuiToolbar-root-223 MuiToolbar-regular-225 MuiToolbar-gutters-224"
    >
      <div
        className="MuiGrid-container-227"
      >
        <div
          className="MuiGrid-item-228 MuiGrid-grid-xs-12-267"
        >
          <a
            className="SimpleAppBar-link-186"
            href="/"
            onClick={[Function]}
          >
            <h4
              className="MuiTypography-root-324 MuiTypography-h4-339 MuiTypography-colorInherit-353 SimpleAppBar-nameLarge-185"
            >
              Chance The Trainer
            </h4>
            <h6
              className="MuiTypography-root-324 MuiTypography-subtitle2-343 MuiTypography-colorInherit-353 SimpleAppBar-nameSmall-184"
            >
              Chance The Trainer sm
            </h6>
          </a>
        </div>
        <div
          className="MuiGrid-item-228 MuiGrid-grid-xs-12-267"
        >
          <h6
            className="MuiTypography-root-324 MuiTypography-h6-341 MuiTypography-colorInherit-353 SimpleAppBar-nameLarge-185"
          >
            by: ajDevs
          </h6>
          <span
            className="MuiTypography-root-324 MuiTypography-caption-334 MuiTypography-colorInherit-353 SimpleAppBar-nameSmall-184"
          >
            by: ajDevs sm
          </span>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`renders demo button correctly 1`] = `
<div
  className="ProgressBar"
>
  <div>
    <label />
  </div>
</div>
`;
